backend/scripts/location_differ.py

```diff
@@ -0,0 +1,300 @@
+                            ] = False
+                            server_data["features"][i]["properties"][
+                                "last_updated"
+                            ] = today
+                        changes += 1  # track that we changed this machine
+                        depr += 1
+                        match = True  # machine was found in existing dict
+                        break  # to not change a machine found in both dicts twice
+
+                    elif cur_state == "retired" and this_state == "unvisited":
+                        logger.debug(f"{this_title} is available again")
+                        # A machine documented as retired is available again
+                        if name == "Device":
+                            # Easy case, we just add this machine to server_dict
+                            if len(device_dict[this_link]) > 1:
+                                logger.warning(
+                                    "A url linking to multiple machines became"
+                                    f"available check manually: {device_dict[this_link]}"
+                                )
+                            new_entry = device_dict[this_link]
+                            for entry in new_entry:
+                                entry["properties"]["status"] = "unvisited"
+                                entry["properties"]["active"] = True
+                                entry["properties"]["last_updated"] = today
+                            server_data["features"].extend(new_entry)
+                        elif name == "Server":
+                            # Machine is already documented in server_dict
+                            idx = server_keys.index(this_link)
+                            # Re-activate machine
+                            server_data["features"][idx]["properties"][
+                                "status"
+                            ] = "unvisited"
+                            server_data["features"][idx]["properties"][
+                                "active"
+                            ] = True
+                            server_data["features"][i]["properties"][
+                                "last_updated"
+                            ] = today
+
+                        changes += 1  # track that we changed this machine
+                        new += 1
+                        match = True  # machine was found in existing dict
+                        break  # to not change a machine found in both dicts twice
+
+                    else:
+                        raise ValueError(
+                            f"Unknown state combinations: {cur_state}, {this_state}"
+                        )
+
+            if match:
+                continue
+
+            # This is a new machine since the key was not found in both dicts
+            if this_state in UNAVAILABLE_MACHINE_STATES:
+                # Untracked machine that is not available, hence we can skip
```

Comment:
Maybe we should add these as well, sometimes they are just temporarily unavailable and it would still be nice to see the grey pin

Reply:
the machine would be added once it becomes available and then the pin would always remain, even if it becomes unavailable